src/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1464,5 +1519,17 @@ private static string DateToString(DateTime D)
             DateTimeFormatInfo dateFormat = new DateTimeFormatInfo();
             return D.ToUniversalTime().ToString("R", dateFormat);
         }
+
+        private bool TryGetHostUri(string hostName, out Uri hostUri)
+        {
+            // Use StringBuilder rather than UriBuilder, since we don't know if the "hostName" will contain a
+            // port or not.
+            StringBuilder sb = new StringBuilder(Address.Scheme);
+            sb.Append("://");
+            sb.Append(hostName);
+            sb.Append(Address.PathAndQuery);
+
+            return Uri.TryCreate(sb.ToString(), UriKind.Absolute, out hostUri);
+        }
```

Comment:
Ok.  This would be better anyway as:
```C#
string s = Address.Scheme + "://" + hostname + Address.PathAndQuery;
return Uri.TryCreate(s, UriKind.Absolute, out hostUri);
```